cmd/bfgd/bfgd.go

```diff
@@ -148,63 +52,82 @@ func HandleSignals(ctx context.Context, cancel context.CancelFunc, callback func
 
 	select {
 	case <-ctx.Done():
-	case s := <-signalChan: // First signal, cancel context.
+	case s := <-signalChan:
 		if callback != nil {
-			callback(s) // Do whatever caller wants first.
-			cancel()
+			callback(s)
 		}
+		cancel()
 	}
-	<-signalChan // Second signal, hard exit.
+	<-signalChan
+	log.Errorf("Received second termination signal, forcing exit.")
 	os.Exit(2)
 }
 
-func _main() error {
-	// Parse configuration from environment
-	if err := config.Parse(cm); err != nil {
-		return err
-	}
-
-	loggo.ConfigureLoggers(cfg.LogLevel)
-	log.Infof(welcome)
+// initializeLogger configures logging for the application.
+func initializeLogger(logLevel string) {
+	loggo.ConfigureLoggers(logLevel)
+	log.Infof("Logger initialized with level: %s", logLevel)
+}
 
-	pc := config.PrintableConfig(cm)
-	for k := range pc {
-		log.Infof("%v", pc[k])
+// parseConfig parses environment variables into the configuration map.
+func parseConfig() error {
+	if err := config.Parse(cm); err != nil {
+		return fmt.Errorf("failed to parse configuration: %w", err)
 	}
+	return nil
+}
```

Comment:
I don't think this function largely unnecessary, and could add unnecessary context to the error. Would you mind inlining `config.Parse` instead?

Reply:
check new commit pls